Add on action prop (#538)
measuredco/puck: apps/docs/pages/docs/api-reference/components/puck.mdx

| [`initialHistory`](#initialhistory) | `initialHistory: {}`                   | [InitialHistory](#initialhistory-params)           | -            |
| [`onChange()`](#onchangedata)       | `onChange: (data) => {}`               | Function                                           | -            |
| [`onPublish()`](#onpublishdata)     | `onPublish: async (data) => {}`        | Function                                           | -            |

> | [`onAction()`](#onactionaction-appstate-prevappstate) | `onAction: (action, appState, prevAppState) => {}`        | Function                                           | -            |

Reviewer: Could you make this alphabetical please :)

The index of the histories to set the user to.

### `onAction(action, appState, prevAppState)`

> Callback that triggers when Puck dispatches an [action](https://puckeditor.com/docs/api-reference/actions), like as `insert` or `set`. Use this to track changes, perform side effects, or sync with external systems.

Reviewer: Sorry, I made a typo in the above. Would you be able to push one more fix?

```suggestion
Callback that triggers when Puck dispatches an [action](https://puckeditor.com/docs/api-reference/actions), like `insert` or `set`. Use this to track changes, perform side effects, or sync with external systems.
```

Author: Sure 😊 I should have checked it.

}
```

> ### `onAction(action, appState, prevAppState)`

Reviewer: Alphabetical please :)

> Callback that triggers when an action occurs in the Puck. Use this to track changes, perform side effects, or sync with external systems.

Reviewer: ```suggestion
Callback that triggers when Puck dispatches an [action](https://puckeditor.com/docs/api-reference/actions), like as `insert` or `set`. Use this to track changes, perform side effects, or sync with external systems.
```

Reviewer: Sorry I made a tiny typo in the above. Could you push one more fix?

```suggestion
Callback that triggers when Puck dispatches an [action](https://puckeditor.com/docs/api-reference/actions), like `insert` or `set`. Use this to track changes, perform side effects, or sync with external systems.
```
